### PR TITLE
FIXED: Chemistry JSON dump tool and companion GitHub Action

### PR DESCRIPTION
**Pull request by weaversam8**
This PR is a followup from #6134, which had to be reverted after we discovered sandbox violations. For full details on the contents of this PR, please review the original linked above. The docs here will focus on changes since #6134.

- This PR now depends on engine PR space-wizards/RobustToolbox#2435.
- The `UniversalJsonConverter<T>` class has been moved from `Content.Shared` to `Content.Server`
- The `UniversalJsonConverter<T>` class is no longer registered via the `[JsonConverter(...)]` attribute but is instead registered via the [`JsonSerializerOptions.Converters` property](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.converters?view=net-6.0#system-text-json-jsonserializeroptions-converters).

A full diff of changes since #6134 is visible [here](https://github.com/weaversam8/space-station-14/compare/03751d2086670dfffbcee7a3f90707797e910716...weaversam8:jsondump) (minus the fact that recent commits from `space-wizards/space-station-14/master` were also merged in since then, in order to bring the `jsondump` branch up to date.)

**Changelog**

:cl: Weaver
- add: The game wiki (wiki.ss14.io)'s chemistry guide is now 100% up to date and automated. You can trust that it's correct now.

### CODE REVIEW

**Review by mirrorcult — APPROVED** [review]

**Comment by moonheart08** [comment]
Needs a submodule update.

File: Content.Server/Entry/EntryPoint.cs [hunk 1]
```diff
@@ -62,45 +68,66 @@ public override void Init()
 
             IoCManager.BuildGraph();
             factory.GenerateNetIds();
+            var configManager = IoCManager.Resolve<IConfigurationManager>();
+            var dest = configManager.GetCVar(CCVars.DestinationFile);
+            if (dest == "") //hacky but it keeps load times for the generator down.
```
Comment (ShadowCommander) [inline Content.Server/Entry/EntryPoint.cs:73]:
```suggestion
            if (string.IsNullOrEmpty(dest)) //hacky but it keeps load times for the generator down.
```

File: Content.Server/Entry/EntryPoint.cs [hunk 2]
```diff
@@ -62,45 +68,66 @@ public override void Init()
...
-            IoCManager.Resolve<IGameMapManager>().Initialize();
-            IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<GameTicker>().PostInitialize();
-            IoCManager.Resolve<IBqlQueryManager>().DoAutoRegistrations();
+            var configManager = IoCManager.Resolve<IConfigurationManager>();
+            var resourceManager = IoCManager.Resolve<IResourceManager>();
+            var dest = configManager.GetCVar(CCVars.DestinationFile);
+            var resPath = new ResourcePath(dest).ToRootedPath();
+            if (dest != "")
+            {
+                var file = resourceManager.UserData.OpenWriteText(resPath.WithName("chem_" + dest));
```
Comment (ShadowCommander) [inline Content.Server/Entry/EntryPoint.cs:106]:
```suggestion
            if (!string.IsNullOrEmpty(dest))
            {
                var resPath = new ResourcePath(dest).ToRootedPath();
                var file = resourceManager.UserData.OpenWriteText(resPath.WithName("chem_" + dest));
```

File: Content.Server/Entry/EntryPoint.cs [hunk 3]
```diff
@@ -110,11 +137,11 @@ public override void Update(ModUpdateLevel level, FrameEventArgs frameEventArgs)
...
-                {
-                    _euiManager.SendUpdates();
-                    _voteManager.Update();
-                    break;
-                }
+                    {
+                        _euiManager.SendUpdates();
+                        _voteManager.Update();
+                        break;
+                    }
```
Comment (ShadowCommander) [inline Content.Server/Entry/EntryPoint.cs:144]:
```suggestion
                {
                    _euiManager.SendUpdates();
                    _voteManager.Update();
                    break;
                }
```
Should be less indent to be consistent with the rest of the codebase.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

**Comment by weaversam8** [comment]
Submodule update from earlier this morning merged in from master. This should be ready to merge.

**Review by moonheart08 — APPROVED** [review]

**Review by ShadowCommander — APPROVED** [review]

**Comment by ShadowCommander** [comment]
Just need someone to test this and make sure it doesn't break things anymore on a live server and then it should be good to merge.

**Comment by moonheart08** [comment]
> Just need someone to test this and make sure it doesn't break things anymore on a live server and then it should be good to merge.

I have done this locally, sandbox no longer dies and the game runs.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**